csrc/includes/cpu_adagrad.h

```diff
@@ -12,21 +12,29 @@
 #include <cassert>
 #include "simd.h"
 
+#ifndef HALF_DTYPE
+#error Must provide compiler option -DHALF_DTYPE=<half data type>
+#endif
+
 #if defined(__ENABLE_CUDA__)
 #include <cuda_fp16.h>
+#ifdef BF16_AVAILABLE
+#include <cuda_bf16.h>
+#endif
 #include <cuda_runtime_api.h>
 #include "cuda.h"
 #include "custom_cuda_layers.h"
-typedef __half ds_half_precision_t;
 #elif defined(__ENABLE_CANN__)
 #include "acl/acl.h"
 #include "torch_npu/csrc/core/npu/NPUStream.h"
-typedef c10::Half ds_half_precision_t;
 #else
 typedef unsigned short ds_half_precision_t;
 #endif
 
+typedef HALF_DTYPE ds_half_precision_t;
```

Comment:
Does this line 34 make line 31 above dead?

Reply:
fixed